gettext-extraction/src/lib.rs

```diff
@@ -0,0 +1,67 @@
+extern crate proc_macro;
+use proc_macro::TokenStream;
+use std::{ffi::OsString, fs::OpenOptions, io::Write};
+
+fn append_po_entry_to_file(message: &TokenStream, file_name: &OsString) {
+    let mut file = OpenOptions::new()
+        .create(true)
+        .append(true)
+        .open(file_name)
+        .unwrap_or_else(|e| panic!("Could not open file {file_name:?}: {e}"));
+    let message_string = message.to_string();
+    if message_string.contains('\n') {
+        panic!("Gettext strings may not contain unescaped newlines. Unescaped newline found in '{message_string}'")
```

Comment:
right, this check is still needed until we translate unescaped newlines to valid msgstr entries